sdk/tables/azure-data-tables/azure/data/tables/_base_client.py

```diff
@@ -392,12 +393,9 @@ def format_query_string(sas_token, credential):
     if sas_token and isinstance(credential, AzureSasCredential):
         raise ValueError(
             "You cannot use AzureSasCredential when the resource URI also contains a Shared Access Signature.")
-    if sas_token and not credential:
-        query_str += sas_token
-    elif is_credential_sastoken(credential):
-        query_str += credential.lstrip("?")
-        credential = None
-    return query_str.rstrip("?&"), credential
+    elif isinstance(credential, AzureSasCredential) or isinstance(credential, AzureNamedKeyCredential):
+        return query_str.rstrip("?"), credential
```

Comment:
I think this should just be:
`return "", credential`?

Reply:
i believe that is what happens, but this is clearer